File: 5Days-SoftwareArchitecture/typesript/Day2/Connascence/con_meaning_good.ts
type TxResult = "SUCCESS" | "INSUFFICIENT_FUNDS" | "ERROR";

class AccountServiceGood {
    withdraw(amount: number): TxResult {
        return "INSUFFICIENT_FUNDS";
    }
}

class ClientMeaningGood {
    private svc = new AccountServiceGood();

    attempt() {
        const result = this.svc.withdraw(500);
        if (result === "INSUFFICIENT_FUNDS")
            console.log("Insufficient funds (explicit)");
    }
}

new ClientMeaningGood().attempt();

File: 5Days-SoftwareArchitecture/Java/Day 2/Connascence/CoN_Meaning_Good.java
enum TxStatus { SUCCESS, INSUFFICIENT_FUNDS, ERROR }

class AccountServiceGood {
    TxStatus withdraw(double amount) {
        return TxStatus.INSUFFICIENT_FUNDS;
    }
}

class ClientMeaningGood {
    private AccountServiceGood service = new AccountServiceGood();

    void attempt() {
        TxStatus status = service.withdraw(500);
        if (status == TxStatus.INSUFFICIENT_FUNDS) {
            System.out.println("Insufficient funds (explicit enum)");
        }
    }
}

public class CoN_Meaning_Good {
    public static void main(String[] args) {
        new ClientMeaningGood().attempt();
    }
}
